dask/array/random.py

```diff
@@ -348,6 +348,48 @@ def random_sample(self, size=None, chunks="auto"):
     def rayleigh(self, scale=1.0, size=None, chunks="auto"):
         return self._wrap('rayleigh', scale, size=size, chunks=chunks)
 
+    def shuffle(self, x):
```

Comment:
Why not call this `permutation` instead, which is consistent with the naming in numpy.random?

Reply:
Oh, I'd forgotten that permutation can take an array! That's much nicer.

I suspect we can also support the integer variant. I'll pick this up later this week or next.